packages/algob/src/internal/deployer.ts

```diff
@@ -541,4 +608,17 @@ export class DeployerRunMode extends DeployerBasicMode implements Deployer {
       methodName: "deploySSC"
     });
   }
+
+  async updateSSC (
+    sender: algosdk.Account,
+    payFlags: rtypes.TxParams,
+    appId: number,
+    newApprovalProgram: string,
+    newClearProgram: string,
+    flags: rtypes.SSCOptionalFlags
+  ): Promise<SSCInfo> {
+    throw new BuilderError(ERRORS.BUILTIN_TASKS.DEPLOYER_EDIT_OUTSIDE_DEPLOY, {
+      methodName: "updateSSC"
+    });
+  }
```

Comment:
I think user should be able to update SSC outside *DEPLOY* mode as well

Reply:
updating is same as deploying. why user can update SSC outside DEPLOY using algob ?

Reply:
Why not? I think updating is not same as setup/deploying.. It could happen that after a while user decides to update the application, so he could add a script in *RUN* mode. Say if i wanted to update existing rules (eg. adding a third rule) in security token, then maybe user does so in run mode? What do you think?

If not, could you please check [https://github.com/scale-it/algo-builder/pull/337/files#diff-84170b6018e0919812f4a60f9c179f1e9d86a04d789d02ac02e2bba865cc23bbR9](https://github.com/scale-it/algo-builder/pull/337/files#diff-84170b6018e0919812f4a60f9c179f1e9d86a04d789d02ac02e2bba865cc23bbR9)? It's using 
 *updateSSC* in *RUN* mode

Reply:
Then what do we do with checkpoints ?
App will be updated but checkpoint will be old, this is not an interaction only function, it can completely change the code, which is same as deploying.

Do you think we should allow deploy in run mode ? According to your comment we should

Reply:
In previous code updatessc function was independent and wasn't associated with checkpoints but once we have it in deployer class, we have to think about checkpoints as well

Reply:
Updating a application is part of re-setup, in most of the cases there will be no application which is modified at each call.

It will be required to be updated once in a while.

Reply:
yes, you are right about that script.
but what do we do with checkpoints ?
`algob deploy scripts/interaction/update_application.js` maybe we can support this command ?

Reply:
> Then what do we do with checkpoints ? but what do we do with checkpoints ?

This is what i was discussing on discord. We make it clear in the documentation that in this case checkpoint is not created. This is a user problem, but i don't think we should take control away from the user. User can do anything in run mode which is not part of "setup". transfer, opt-in, udpate.. 

> Do you think we should allow deploy in run mode ? According to your comment we should

No. User will only deploy an app once, but it could be updated multiple times (maybe dynamically in a script using template parameter). Updating is different than deploying. 

> App will be updated but checkpoint will be old, this is not an interaction only function, it can completely change the code, which is same as deploying.

We can add a function in deployer to query the blockchain to retreive app info (in case user updates an application in *RUN* mode), what do you think? [https://github.com/algorand/js-algorand-sdk/blob/develop/src/client/v2/algod/getApplicationByID.ts](https://github.com/algorand/js-algorand-sdk/blob/develop/src/client/v2/algod/getApplicationByID.ts)

Reply:
I am not not sure about moving update ssc to readmode.